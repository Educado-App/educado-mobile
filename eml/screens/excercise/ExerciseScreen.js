import { StatusBar } from "expo-status-bar";
import React, { useEffect, useRef, useState } from "react";
import { Alert, StyleSheet, View, Text } from "react-native";
import LeaveButton from "../../components/exercise/LeaveButton";
import ExerciseButtons from "../../components/exercise/ExerciseButtons";
import { useNavigation, useRoute } from "@react-navigation/native";
import { Video } from "expo-av";
import * as StorageService from "../../services/StorageService";
import CustomProgressBar from "../../components/exercise/Progressbar";
import dummyExerciseData from './dummyExerciseData.json';



export default function ExerciseScreen() {
  const navigation = useNavigation();
  const route = useRoute();
  
  const [hasData, setHasData] = useState(false);
  const [signal, setSignal] = useState([]);
  const [exerciseData, setExerciseData] = useState({});

  /*async function getExercise() {
    const exercise = await StorageService.getNextExercise(sectionId);

    if (exercise !== null) {
      console.log(exercise);

      if (exercise === true) {
        navigation.navigate("SectionComplete", {
          courseId: courseId,
          sectionId: sectionId,
        });
      } else {
        setExerciseData(exercise);
      }
    } else {
      navigation.navigate("ErrorScreen");
    }
  }*/

  async function getExercise() {
    // Instead of calling StorageService.getNextExercise, use the dummy data
    const exercise = dummyExerciseData;
  
    if (exercise !== null) {
      console.log(exercise);
  
     /* const courseId = exercise.courseId || "defaultCourseId";
      const sectionId = exercise.sectionId || "defaultSectionId";*/
  
      if (exercise === true) {
       /* navigation.navigate("SectionComplete", {
          courseId: courseId,
          sectionId: sectionId,
        });*/
      } else {
        setExerciseData(exercise);
      }
    } else {
      navigation.navigate("ErrorScreen");
    }
  }
  
  

  /*async function isSectionComplete(courseId, sectionId) {
    const course = await StorageService.getCourseById(courseId);
    const sections = course.sections;

    for (let i = 0; i < sections.length; i++) {
      console.log("Section is complete: ", sections[i].isComplete);
      if (sections[i].id === sectionId) {
        return sections[i].isComplete;
      }
    }
  }*/

  useEffect(() => {
    getExercise().then(() => {
      setHasData(true);
    });
  }, [route.params]);

  const video = useRef(0);

  useEffect(() => {
    if (video !== undefined) {
      if (signal === 0) {
        try {
          video.current.pauseAsync();
        } catch (e) {
          console.log("Something went wrong :" + e);
        }
      }
      setSignal(1);
    }
  }, [signal]);

  return (
    <View style={styles.container} className="bg-babyBlue">
      <View style={{ flex: 1 }}>
        <View className="pt-8 flex-row w-screen">
          <View className="">
            <LeaveButton
              //navigationPlace={"Course"}
              //courseId={courseId}
            ></LeaveButton>
          </View>
          <View
            style={{
              left: "80%",
              justifyContent: "center",
              alignItems: "center",
              flexDirection: "column",
            }}
          ></View>
        </View>
        <View className="items-center">
          <CustomProgressBar progress={0.5 / 1}></CustomProgressBar>
        </View>
      </View>
      <View style={{ flex: 2, width: "100%" }}>
        {exerciseData === undefined ? (
          // No Data
          <Text> Sem dados</Text>
        ) : (
          <Video
            source={{ uri: exerciseData.content }}
            rate={1.0}
            volume={1.0}
            isMuted={false}
            resizeMode="cover"
            shouldPlay
            useNativeControls
            ref={video}
            style={styles.backgroundVideo}
          />
        )}
      </View>
      <View style={{ flex: 3 }}>
        {exerciseData === undefined ? (
          //No data
          <Text> Sem dados</Text>
        ) : (
          <ExerciseButtons
            answers={exerciseData.answers}
            exerciseId={exerciseData.id}
            courseId={exerciseData.courseId}
            sectionId={exerciseData.sectionId}
            setSignal={setSignal}
            hasData={hasData}
          ></ExerciseButtons>
        )}
      </View>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
  },
  row: {
    flexDirection: "row",
  },
  textStyle: {
    zIndex: 100,
    fontSize: 100,
  },
  buttonShadow: {
    shadowOpacity: 0.4,
    shadowRadius: 10,
    shadowOffset: { width: 0, height: 0 },
  },
  backgroundVideo: {
    height: "100%",
  },
});
