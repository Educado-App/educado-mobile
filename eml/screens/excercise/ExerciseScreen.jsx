import { StatusBar } from 'expo-status-bar';
import React, { useEffect, useState } from 'react';
import { ScrollView, StyleSheet, View, Image, TouchableHighlight, Pressable, TouchableOpacity, Dimensions, SafeAreaView } from "react-native";
import { useNavigation, useRoute } from '@react-navigation/native';
import LeaveButton from '../../components/exercise/LeaveButton';
import * as StorageService from '../../services/StorageService';
import Text from '../../components/general/Text';
import CustomProgressBar from "../../components/exercise/Progressbar";
import dummyExerciseData from "./dummyExerciseData.json";
import { Button, RadioButton } from "react-native-paper";
import ExerciseInfo from "../../components/exercise/ExerciseInfo";
import { ScreenWidth } from "@rneui/base";
import { Icon } from '@rneui/themed';

export default function ExerciseScreen() {
  const navigation = useNavigation();
  const route = useRoute();
  const screenHeight = Dimensions.get('window').height;
  const tailwindConfig = require('../../tailwind.config.js');
  const projectColors = tailwindConfig.theme.colors;

  const [hasData, setHasData] = useState(false);
  const [signal, setSignal] = useState([]);
  const [exerciseData, setExerciseData] = useState({});
  const [selectedAnswer, setSelectedAnswer] = useState(null); // State to store the selected answer
  const [buttonClassName, setButtonClassName] = useState(""); // Used to change color of a view
  const [showFeedback, setShowFeedback] = useState(false); // Used to render feedback


  const handleAnswerSelect = (answerId) => {
    setSelectedAnswer(answerId);
  };

  var reviewAnswer;

  // Update this function to look like handleAnswerSelect, looks better
  function handleReviewAnswer() {
    if (dummyExerciseData.answers[selectedAnswer - 1].isCorrect) {
      setButtonClassName("bg-projectGreen");
      reviewAnswer = true;
    } else {
      setButtonClassName("bg-projectRed");
      reviewAnswer = false;
    }
    setShowFeedback(true);
    console.log(reviewAnswer);
  }

  /*async function getExercise() {
    const exercise = await StorageService.getNextExercise(sectionId);

    if (exercise !== null) {
      console.log(exercise);

      if (exercise === true) {
        navigation.navigate("SectionComplete", {
          courseId: courseId,
          sectionId: sectionId,
        });
      } else {
        setExerciseData(exercise);
      }
    } else {
      navigation.navigate("ErrorScreen");
    }
  }*/

  async function getExercise() {
    // Instead of calling StorageService.getNextExercise, use the dummy data
    const exercise = dummyExerciseData;

    if (exercise !== null) {
      /* const courseId = exercise.courseId || "defaultCourseId";
      const sectionId = exercise.sectionId || "defaultSectionId";*/

      if (exercise === true) {
        /* navigation.navigate("SectionComplete", {
          courseId: courseId,
          sectionId: sectionId,
        });*/
      } else {
        setExerciseData(exercise);
      }
    } else {
      navigation.navigate("ErrorScreen");
    }

  }

  /*async function isSectionComplete(courseId, sectionId) {
    const course = await StorageService.getCourseById(courseId);
    const sections = course.sections;

    for (let i = 0; i < sections.length; i++) {
      console.log("Section is complete: ", sections[i].isComplete);
      if (sections[i].id === sectionId) {
        return sections[i].isComplete;
      }
    }
  }*/

  useEffect(() => {
    getExercise().then(() => {
      setHasData(true);
    });
  }, [route.params]);

  return (
    <View className="bg-secondary flex-1 justify-between">
      <SafeAreaView className="justify-between" >
        <View className="flex-row items-center justify-around">
          <View>
            <LeaveButton
              navigationPlace={"Course"}
              courseId={dummyExerciseData.courseId}
            ></LeaveButton>
          </View>
          <View>
            <CustomProgressBar progress={0.25 / 1}></CustomProgressBar>
          </View>
          <View>
            <Text className="px-3 text-center font-montserrat-bold text-caption-medium text-projectBlack">
              25%
            </Text>
          </View>
        </View>

        {dummyExerciseData === undefined ? (
          // No Data
          <Text> Sem dados</Text>
        ) : (
          <View className="items-center">
            <Text className="pt-6 pb-10 text-center text-body font-montserrat-bold text-projectBlack w-5/6">
              {dummyExerciseData.question}
            </Text>
            <View className={`${buttonClassName} items-center justify-center`} style={{ height: screenHeight * 0.51, width: ScreenWidth * 1 }}>
              <ScrollView>
                {/* Map through the answers and render each one */}
                {dummyExerciseData.answers.map((answer) => (
                  <View
                    key={answer.id}
                    className="flex-row w-[390] pb-6 pl-2"
                  >
                    <View>
                      <RadioButton.Android
                        value={answer.id}
                        status={
                          selectedAnswer === answer.id ? "checked" : "unchecked"
                        }
                        onPress={() => handleAnswerSelect(answer.id)}
                        color="#5ECCE9"
                        uncheckedColor="#5ECCE9"
                      />
                    </View>
                    <View>
                      <Pressable onPress={() => handleAnswerSelect(answer.id)}>
                        <Text className="pt-2 pb-1 w-[304] font-montserrat text-body text-projectBlack">{answer.text}</Text>
                      </Pressable>
                      {showFeedback ? (
                        <View className={`flex-row pb-2 w-[310] rounded ${answer.isCorrect ? 'bg-projectGreen' : 'bg-projectRed'}`}>
                          <View className="pl-2 pt-1">
                            <View className="pt-1.5">
                              {answer.isCorrect === true ? (
                                <Icon
                                  size={10}
                                  name="check"
                                  type="material"
                                  color={projectColors.success}
                                />
                              ) : (
                                <Icon
                                  size={10}
                                  name="close"
                                  type="material"
                                  color={projectColors.error}
                                />
                              )}
                            </View>
                          </View>
                          <Text className={`w-[272] pl-1 pt-2 font-montserrat text-caption-medium rounded-medium ${answer.isCorrect ? 'text-success' : 'text-error'}`}>{answer.feedback}</Text>
                        </View>
                      ) : null}
                    </View>
                  </View>
                ))}
              </ScrollView>
            </View>
            <View className="px-6 pt-8 w-screen">
              <TouchableOpacity
                className={`${selectedAnswer !== null ? 'opacity-100' : 'opacity-30'} bg-primary px-10 py-4 rounded-medium`}
                onPress={() => handleReviewAnswer()}
              >
                <Text className="text-center font-montserrat-bold text-body text-projectWhite">Confirmar resposta</Text>
              </TouchableOpacity>
            </View>
          </View>
        )}

        {/* Old exercise buttons
        <View style={{ flex: 3 }}>
          {exerciseData === undefined ? (
            //No data
            <Text> Sem dados</Text>
          ) : (
            <ExerciseButtons
              answers={exerciseData.answers}
              exerciseId={exerciseData.id}
              courseId={exerciseData.courseId}
              sectionId={exerciseData.sectionId}
              setSignal={setSignal}
              hasData={hasData}
            ></ExerciseButtons>
          )}
        </View>
        */}

        <ExerciseInfo courseId={dummyExerciseData.courseId} sectionId={dummyExerciseData.sectionId} />
        <StatusBar style="auto" />
      </SafeAreaView>
    </View>
  );
}
