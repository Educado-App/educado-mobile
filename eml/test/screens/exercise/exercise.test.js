import React from 'react';
import renderer from 'react-test-renderer';
import ExerciseScreen from '../../../screens/excercise/ExerciseScreen';
import AsyncStorage from "@react-native-async-storage/async-storage";

let navigated = false;

jest.mock('@react-navigation/native', () => ({
    useNavigation: () => ({
      navigate: jest.fn(() => { navigated = true }),
    }),
    useRoute: () => ({
      params: {}, // Add any route parameters your component relies on
    }),
  }));

let excerciseScreen;

beforeEach(() => {
  navigated = false;
  AsyncStorage.clear();
  excerciseScreen = renderer.create(<ExerciseScreen />);
});

describe('ExerciseScreen', () => {
  it('renders ExerciseScreen correctly', () => {
    expect(excerciseScreen.toJSON()).toMatchSnapshot();
  });


//Answer Selection Test: 
//Write a test to simulate the selection of an answer. 
//Check if the `handleAnswerSelect` function updates the `selectedAnswer` state correctly when a radio button is pressed.

//Rendering Exercise Content Test: 
//Ensure that the exercise content, including the question and answer options, is rendered correctly based on the `dummyExerciseData`.


//Review Answer Test: 
//If the "Review answer" feature is important, write tests to simulate the review action and verify the expected behavior when the review button is pressed.

});
