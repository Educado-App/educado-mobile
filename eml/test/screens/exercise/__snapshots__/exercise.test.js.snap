// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseScreen renders ExerciseScreen correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F1F9FB",
      },
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "justifyContent": "space-between",
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      [
        {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 20,
              },
              undefined,
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "invisible",
                  "borderColor": "#2089dc",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "padding": 4,
                  "paddingHorizontal": 6,
                  "width": 50,
                }
              }
            >
              <View
                style={
                  {
                    "overflow": "hidden",
                  }
                }
                testID="RNE__ICON__CONTAINER"
              >
                <View
                  accessibilityRole="button"
                  android_ripple={
                    {
                      "borderless": false,
                      "color": "rgba(0, 0, 0, 0.3)",
                      "radius": -5,
                    }
                  }
                  disabled={false}
                  testID="RNE__ICON__CONTAINER_ACTION"
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    testID="RNE__ICON"
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          animated={true}
          animationConfig={
            {
              "bounciness": 0,
            }
          }
          animationType="spring"
          indeterminate={false}
          indeterminateAnimationDuration={1000}
          onLayout={[Function]}
          progress={0.25}
          style={
            [
              {
                "backgroundColor": "rgba(228, 242, 245, 1)",
                "borderColor": "rgba(94, 204, 233, 1)",
                "borderRadius": 8,
                "borderWidth": 0,
                "overflow": "hidden",
                "width": 253,
              },
              undefined,
            ]
          }
          useNativeDriver={false}
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(94, 204, 233, 1)",
                "height": 10,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateX": -94.875,
                  },
                  {
                    "scaleX": 0.250075,
                  },
                ],
              }
            }
          />
        </View>
      </View>
      <View>
        <Text
          style={
            [
              {
                "paddingLeft": 12,
                "paddingRight": 12,
              },
              {
                "textAlign": "center",
              },
              {
                "fontFamily": "Montserrat-Bold",
              },
              {
                "fontSize": 12,
              },
              {
                "color": "#383838",
              },
            ]
          }
        >
          25%
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "paddingTop": 24,
            },
            {
              "paddingBottom": 40,
            },
            {
              "textAlign": "center",
            },
            {
              "fontSize": 16,
            },
            {
              "fontFamily": "Montserrat-Bold",
            },
            {
              "color": "#383838",
            },
            {
              "width": "83.333333%",
            },
          ]
        }
      >
        Qual é a pergunta que o content creator irá fazer nessa etapa?
      </Text>
      <View
        style={
          [
            {
              "alignItems": "center",
            },
            {
              "justifyContent": "center",
            },
            {
              "height": 680.34,
              "width": 750,
            },
          ]
        }
      >
        <RCTScrollView>
          <View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "width": 390,
                  },
                  {
                    "paddingBottom": 24,
                  },
                  {
                    "paddingLeft": 8,
                  },
                ]
              }
            >
              <View>
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="radio"
                  accessibilityState={
                    {
                      "checked": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 18,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "borderColor": "#5ECCE9",
                        "borderRadius": 10,
                        "borderWidth": 2,
                        "height": 20,
                        "margin": 8,
                        "width": 20,
                      }
                    }
                  />
                </View>
              </View>
              <View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <Text
                    style={
                      [
                        {
                          "paddingTop": 8,
                        },
                        {
                          "paddingBottom": 4,
                        },
                        {
                          "width": 304,
                        },
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "#383838",
                        },
                      ]
                    }
                  >
                    Option option option option option option placeholder  placeholder placeholder
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "width": 390,
                  },
                  {
                    "paddingBottom": 24,
                  },
                  {
                    "paddingLeft": 8,
                  },
                ]
              }
            >
              <View>
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="radio"
                  accessibilityState={
                    {
                      "checked": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 18,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "borderColor": "#5ECCE9",
                        "borderRadius": 10,
                        "borderWidth": 2,
                        "height": 20,
                        "margin": 8,
                        "width": 20,
                      }
                    }
                  />
                </View>
              </View>
              <View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <Text
                    style={
                      [
                        {
                          "paddingTop": 8,
                        },
                        {
                          "paddingBottom": 4,
                        },
                        {
                          "width": 304,
                        },
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "#383838",
                        },
                      ]
                    }
                  >
                    Option option option option option option placeholder  placeholder placeholder
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "width": 390,
                  },
                  {
                    "paddingBottom": 24,
                  },
                  {
                    "paddingLeft": 8,
                  },
                ]
              }
            >
              <View>
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="radio"
                  accessibilityState={
                    {
                      "checked": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 18,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "borderColor": "#5ECCE9",
                        "borderRadius": 10,
                        "borderWidth": 2,
                        "height": 20,
                        "margin": 8,
                        "width": 20,
                      }
                    }
                  />
                </View>
              </View>
              <View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <Text
                    style={
                      [
                        {
                          "paddingTop": 8,
                        },
                        {
                          "paddingBottom": 4,
                        },
                        {
                          "width": 304,
                        },
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "#383838",
                        },
                      ]
                    }
                  >
                    Option option option option option option placeholder  placeholder placeholder
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "width": 390,
                  },
                  {
                    "paddingBottom": 24,
                  },
                  {
                    "paddingLeft": 8,
                  },
                ]
              }
            >
              <View>
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="radio"
                  accessibilityState={
                    {
                      "checked": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 18,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "borderColor": "#5ECCE9",
                        "borderRadius": 10,
                        "borderWidth": 2,
                        "height": 20,
                        "margin": 8,
                        "width": 20,
                      }
                    }
                  />
                </View>
              </View>
              <View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <Text
                    style={
                      [
                        {
                          "paddingTop": 8,
                        },
                        {
                          "paddingBottom": 4,
                        },
                        {
                          "width": 304,
                        },
                        {
                          "fontFamily": "Montserrat-Regular",
                        },
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "#383838",
                        },
                      ]
                    }
                  >
                    Option option option option option option placeholder  placeholder placeholder
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          [
            {
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            {
              "paddingTop": 32,
            },
            {
              "width": 750,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#5ECCE9",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "opacity": 0.3,
              "paddingBottom": 16,
              "paddingLeft": 40,
              "paddingRight": 40,
              "paddingTop": 16,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontFamily": "Montserrat-Bold",
                },
                {
                  "fontSize": 16,
                },
                {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Confirmar resposta
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "alignItems": "flex-end",
          },
          {
            "paddingLeft": "6%",
            "paddingRight": "6%",
          },
          {
            "paddingBottom": "12%",
            "paddingTop": "12%",
          },
        ]
      }
    >
      <View>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "fontSize": 10,
              },
              {
                "color": "#A1ACB2",
              },
            ]
          }
        >
          Course name: 
          lecture 1
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Bold",
              },
              {
                "fontSize": 16,
              },
              {
                "color": "#383838",
              },
            ]
          }
        >
          Question title
        </Text>
      </View>
      <View>
        <Image
          source={1}
        />
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
